mono/metadata/icall.c

```diff
@@ -160,29 +162,46 @@ mono_icall_get_file_path_prefix (const gchar *path)
 }
 #endif /* HOST_WIN32 */
 
-ICALL_EXPORT MonoObject *
-ves_icall_System_Array_GetValueImpl (MonoArray *arr, guint32 pos)
+ICALL_EXPORT MonoObjectHandle
+ves_icall_System_Array_GetValueImpl (MonoArrayHandle arr, guint32 pos, MonoError *error)
 {
-	ERROR_DECL (error);
+	if (icall_array_verbose)
```

Comment:
if we are going to leave debugging code in, we should at least wrap it in `G_UNLIKELY`

Reply:
I think this is fine. FYI, what the rest of the runtime does is use variadic debug print macros. IE:

```c
#ifdef DEBUG_ICALLS
#define DEBUG_PRINTF(...) do { printf( __VA_ARGS__); } while (0)
#else
#define DEBUG_PRINTF(...) do { } while (0)
#endif
```

Reply:
It depends if you want to keep it being checked by compiler or let it bitrot but granted bitrot would be ok or even preferable in this area. I see some apparent bitroted printfs and then I don't have to further maintain them, i.e. when doing coop version. Could just be ```
#ifdef DEBUG_ICALLS
#define DEBUG_PRINTF printf
#else
#define DEBUG_PRINTF(...) /* nothing */
#endif```

Reply:
old obscure good form:
static bool verbose;
#define PRINT (!verbose) ? (void)0 : g_print
prevents rot, does not evaluate the print usually, but compiler still checks it
trick is the varargs function is at the end of the macro, followed by its parameters, but end of ternary won't be evaluated at runtime, and compiler will optimize it away

Reply:
Stay consistent with the rest of the codebase.